### PR TITLE
Api State Removal - backend

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Always require sm_secret_namespace and sm_secret_name URL parameters for API calls. Add an interface to test API calls that require SMClient

Changes proposed in this pull request:
* Remove states from backend
* Adjust tests
* Adjust UI for changes

**Related issue(s)**

See #442

### CODE REVIEW

File: ui/src/components/ServiceInstancesView.tsx [hunk 1]
```diff
@@ -45,40 +43,25 @@ function ServiceInstancesView(props: any) {
...
+            if (id) {
+              const instance = response.data.items.find((instance) => instance.id === id);
+              if (instance) {
+                openPortal(instance);
+              }
+            }
+            setLoading(false);
           })
           .catch((error) => {
+            setServiceInstances(error);
```
Comment (ralikio) [inline ui/src/components/ServiceInstancesView.tsx:64]: Setting error instances is not correct. Error is not a list of service instances.

**Review by ralikio — CHANGES_REQUESTED** [review]
Review remarks

File: ui/src/components/ServiceInstancesView.tsx (hunk 1 above)
Comment (MarekMichali) [reply ui/src/components/ServiceInstancesView.tsx]: Changed to undefined.

File: internal/api/api.go [hunk 2]
```diff
@@ -65,18 +80,15 @@ func (a *API) Start() {
...
 	router.HandleFunc("POST /api/service-instances", a.CreateServiceInstance)
-	router.HandleFunc("PATCH /api/service-instances/{id}", a.UpdateServiceInstance)
-	router.HandleFunc("DELETE /api/service-instances/{id}", a.DeleteServiceInstance)
-	router.HandleFunc("GET /api/service-bindings", a.ListServiceBindings)
-	router.HandleFunc("GET /api/service-bindings/{id}", a.GetServiceBinding)
+	router.HandleFunc("PATCH /api/service-instances", a.UpdateServiceInstance)
+	router.HandleFunc("DELETE /api/service-instances", a.DeleteServiceInstance)
+	router.HandleFunc("GET /api/service-bindings", a.HandleGetServiceBinding)
 	router.HandleFunc("POST /api/service-bindings", a.CreateServiceBinding)
-	router.HandleFunc("DELETE /api/service-bindings/{id}", a.DeleteServiceBinding)
```
Comment (ralikio) [inline internal/api/api.go:78]: Lest keep ids in the path.

File: internal/api/api.go [hunk 3]
```diff
@@ -65,18 +80,15 @@ func (a *API) Start() {
 
 func (a *API) AttachRoutes(router *http.ServeMux) {
 	router.HandleFunc("GET /api/secrets", a.ListSecrets)
-	router.HandleFunc("GET /api/service-offerings/{namespace}/{name}", a.ListServiceOfferings)
-	router.HandleFunc("GET /api/service-offerings/{id}", a.GetServiceOffering)
-	router.HandleFunc("GET /api/service-instances", a.ListServiceInstances)
-	router.HandleFunc("GET /api/service-instances/{id}", a.GetServiceInstance)
+	router.HandleFunc("GET /api/service-offerings", a.HandleGetServiceOffering)
```
Comment (ralikio) [inline internal/api/api.go:83]: Missing plural.

File: internal/api/api.go (hunk 2 above)
Comment (MarekMichali) [reply internal/api/api.go]: Reverted.

**Review by ralikio — APPROVED** [review]

**Pull request merged**